core/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -70,6 +69,8 @@ public static XContentBuilder builder(XContent xContent, Set<String> includes, S
 
     private boolean humanReadable = false;
 
+    private TimeUnit timeUnit = TimeUnit.MILLISECONDS;
```

Comment:
Allowing this to be customizable is nice but it is going to break parsing pretty hard if this doesn't stay MILLISECONDS.